### ISSUE TITLE
#1 Update PaipuDetail GET method

### ISSUE THREAD

**Issue by ZhuoerFeng** [ZhuoerFeng/TsingMJ#1]
Currently, Paipu log supports only the following contents:

`
[ [ [ 0, 0, 0 ],
        [ 25000, 25000, 25000, 25000 ],
        [ 25 ],
        [ 38 ],
        [ 39, 41, 18, 32, 28, 13, 15, 31, 45, 28, 14, 47, 23 ],
        [ 24, 53, 33, 43, 44, 38, 27, 11, 22, 12, 19, 37 ],
        [ 45, 47, 41, 60, 60, 18, 60, 60, "r53", 60, 60 ],
        [ 44, 12, 16, 27, 25, 17, 29, 17, 44, 11, 39, 42, 27 ],
        [ 35, 39, 42, 15, 44, 14, 21, 21, 18, 38, 26 ],
        [ 27, 17, 11, 12, 35, 17, 60, 60, 60, 44, 44 ],
        [ 37, 11, 19, 26, 12, 22, 34, 23, 28, 14, 14, 24, 16 ],
        [ "c272628", 31, 22, 29, 35, 24, 34, 16, 16, 17, 47 ],
        [ 19, 60, 11, 60, 37, 12, 16, 60, 35, 34, 34 ],
        [ 26, 37, 46, 46, 43, 36, 47, 45, 43, 34, 19, 33, 21 ],
        [ 32, 12, 31, 19, 25, 21, 13, 13, 18, 13, 46 ],
        [ 19, 21, 12, 60, 45, 60, 60, 60, 60, 47, 34 ],
        [ "和了", [ 7000, -2000, -2000, -2000 ],
        [ 0, 0, 0, "30符3飜2000点∀", "Fully Concealed Hand(1飜)", "Riichi(1飜)", "Ura Dora(1飜)" ] ] ],
        [ [ 0, 1, 0 ],
        [ 31000, 23000, 23000, 23000 ],
        [ 14 ],
        [],
        [ 39, 17, 41, 11, 32, 38, 13, 28, 33, 36, 15, 42, 11 ],
        [ 47, 22, 45, 41, 52, 19, 27, 35, 45, 41, 35, 36, 34, 24, 42, 26 ],
        [ 42, 39, 47, 45, 28, 60, 36, 38, 35, 22, 45, 60, 35, 27, 60, 13 ],
        [ 29, 24, 44, 31, 11, 26, 18, 15, 26, 38, 11, 46, 13 ],
        [ 21, 12, 32, 19, 16, 25, 22, 39, 29, 47, 36, 41, 16, 44, 51 ],
        [ 44, 31, 29, 32, 21, 38, 60, 60, 60, 18, 60, 12, 16, 16, 13 ],
        [ 14, 37, 46, 22, 34, 31, 33, 45, 24, 22, 47, 42, 12 ],
        [ 46, 45, 23, "45p4545", "c323334", 53, 13, 46, 25, 23, 27, 12, 16, 42, 32, 47 ],
        [ 42, 31, 37, 22, 14, 60, 60, 12, 22, 23, 60, 60, 60, 60, 60 ],
        [ 13, 29, 18, 23, 38, 27, 23, 18, 44, 28, 21, 28, 17 ],
        [ 25, 33, 34, 16, 19, 27, 39, 36, 32, 15, 24, 12, 33, 43, 31 ],
        [ 44, 28, 38, 21, 60, 18, 60, 60, 13, "r18", 60, 60, 60, 60, 60 ],
        [ "和了", [ -4100, -2100, 9300, -2100 ],
        [ 2, 2, 2, "Mangan 2000-4000点", "Little Three Dragons(2飜)", "Green Dragon(1飜)", "White Dragon(1飜)" ] ] ],
        [ [ 1, 0, 0 ],
        [ 26900, 20900, 32300, 19900 ],
        [ 39 ],
        [ 11 ],
        [ 37, 13, 32, 38, 26, 45, 26, 41, 46, 23, 33, 31, 41 ],
        [ 16, 16, 14, 17, 29, 15, 19, 39, 18 ],
        [ 46, 45, 23, 26, 60, 26, 60, "r16" ],
        [ 17, 13, 27, 29, 16, 11, 37, 33, 45, 27, 51, 11, 36 ],
        [ 36, 42, 45, "4545p45", 15, 23, 21, 19, 36, 47 ],
        [ 33, 60, 36, 29, 27, 60, 60, 27, 19, 60 ],
        [ 31, 13, 13, 38, 24, 45, 46, 32, 11, 34, 43, 44, 41 ],
        [ 12, 25, 38, 22, 29, 21, 38, 18, 25 ],
        [ 44, 43, 45, 46, 41, 29, 34, 60, 21 ],
        [ 42, 44, 34, 21, 34, 52, 28, 29, 23, 46, 28, 42, 24 ],
        [ 21, 26, 33, 12, 28, "21p2121", 53, 37, 41 ],
        [ 44, 34, 46, 60, 29, 34, 60, 60, 33 ],
        [ "和了", [ 8900, -3900, -2000, -2000 ],
        [ 0, 0, 0, "30符4飜2000-3900点", "Fully Concealed Hand(1飜)", "Riichi(1飜)", "Ippatsu(1飜)", "Dora(1飜)", "Ura Dora(0飜)" ] ] ],
        [ [ 2, 0, 0 ],
        [ 34800, 17000, 30300, 17900 ],
        [ 13 ],
        [ 45 ],
        [ 26, 52, 21, 43, 32, 19, 14, 31, 18, 35, 42, 17, 12 ],
        [ 53, 32, 27, 41, 43, 28, 13, 24, 24, 36, 11, 38, 37, 14, 43, 11 ],
        [ 42, 21, 43, 60, 60, 32, "r28", 60, 60, 60, 60, 60, 60, 60, 60, 60 ],
        [ 11, 43, 22, 46, 14, 45, 26, 38, 51, 23, 37, 25, 13 ],
        [ 24, 19, 35, 44, 32, 16, 33, 27, 12, 16, 34, 18, 29, 47, 45, 38 ],
        [ 43, 60, 60, 60, 60, 46, 45, 24, 27, 22, 11, 38, 37, 29, 34, 60 ],
        [ 28, 22, 46, 34, 46, 18, 42, 31, 41, 44, 29, 33, 37 ],
        [ 16, 21, 32, 27, 16, 39, 23, 47, 19, 47, 31, 15, 29, 39, 34, 36, 33 ],
        [ 44, 42, 41, 37, 31, 60, "r18", 60, 60, 60, 60, 60, 60, 60, 60, 60, 60 ],
        [ 18, 15, 44, 19, 39, 23, 15, 42, 36, 26, 22, 17, 38 ],
        [ 25, 47, 28, 21, 39, 11, 41, 46, 23, 29, 24, 12, 13, 44, 17, 21 ],
        [ 44, 39, 47, 42, 60, 60, 60, 28, 21, 19, 36, 15, 38, 15, 29, 44 ],
        [ "和了", [ 10000, 0, -8000, 0 ],
        [ 0, 2, 0, "Mangan 8000点", "Riichi(1飜)", "Dora(1飜)", "Red Five(2飜)", "Ura Dora(0飜)" ] ] ],
        [ [ 3, 0, 0 ],
        [ 43800, 17000, 21300, 17900 ],
        [ 13 ],
        [],
        [ 23, 47, 34, 41, 31, 32, 38, 21, 22, 38, 22, 26, 26 ],
        [ 32, 32, 28, 34, 11, 13, 21, 22, 23 ],
        [ 41, 47, 31, 28, 60, 21, 13, 21, 38 ],
        [ 17, 45, 43, 45, 16, 15, 34, 52, 36, 45, 43, 21, 44 ],
        [ 15, 39, 47, 28, 46, 13, 42, 29, 29 ],
        [ 21, 60, 60, 60, 15, 60, 60, 60, 60 ],
        [ 17, 28, 35, 32, 45, 15, 18, 36, 23, 16, 27, 12, 44 ],
        [ 12, 14, 33, 36, 38, 14, 39, "1414p14", 41, 26 ],
        [ 45, 44, 23, 60, 60, 33, 32, 39, 60, 18 ],
        [ 25, 28, 26, 37, 29, 27, 12, 41, 33, 42, 47, 23, 14 ],
        [ 37, 41, 25, 31, 31, 33, 24, 22, 17, "p414141", 27 ],
        [ 42, 47, 33, 60, 60, 60, 12, 14, 60, 25 ],
        [ "和了", [ -1000, -1000, -1000, 3000 ],
        [ 3, 3, 3, "30符2飜1000点∀", "Prevalent Wind(1飜)", "Seat Wind(1飜)" ] ] ],
        [ [ 3, 1, 0 ],
        [ 42800, 16000, 20300, 20900 ],
        [ 16 ],
        [],
        [ 41, 21, 27, 37, 22, 44, 38, 25, 13, 37, 13, 12, 16 ],
        [ 44, 26, 52, 32, 39, 23, 27, 17, 22, 34, 23, 26, 22, 35, 31 ],
        [ 22, 21, 16, 60, 41, 37, 13, 60, 60, 60, 60, 23, 60, 60, 60 ],
        [ 45, 16, 18, 33, 25, 28, 42, 17, 25, 36, 43, 21, 29 ],
        [ 44, 46, 12, 18, 34, 23, 42, 15, 19, 11, 22, 43, 23 ],
        [ 60, 42, 21, 46, 36, 12, 60, 43, 18, 23, 60, 60, 29 ],
        [ 31, 17, 33, 26, 17, 18, 21, 31, 14, 14, 47, 46, 47 ],
        [ 47, 36, 36, 41, 34, 39, 21, "17p1717", 36, 35, 19, 38, 19, 26 ],
        [ 21, 46, 33, 26, 60, 18, 60, 39, 41, 60, 60, 60, 60, 60 ],
        [ 42, 44, 51, 24, 19, 14, 45, 16, 24, 13, 27, 42, 32 ],
        [ 46, 37, 28, 38, 53, 11, 34, 13, 32, 12, 43, 12, 39, 32, 41 ],
        [ 44, 19, 46, 45, 32, 60, 13, 60, 60, 60, 60, 60, 42, 60, 60 ],
        [ "和了", [ -12300, 0, 12300, 0 ],
        [ 2, 0, 2, "Haneman 12000点", "All Triplets(2飜)", "Red Dragon(1飜)", "Dora(3飜)" ] ] ],
        [ [ 4, 0, 0 ],
        [ 30500, 16000, 32600, 20900 ],
        [ 21 ],
        [ 17 ],
        [ 39, 14, 38, 43, 39, 36, 24, 45, 29, 31, 43, 37, 28 ],
        [ 45, 41, 42, 32, 34, 25, 32, 24, 33 ],
        [ 31, 60, 60, 60, 28, 29, 14, 43, 43 ],
        [ 18, 24, 18, 47, 47, 31, 36, 27, 11, 13, 11, 29, 12 ],
        [ 37, 22, 12, "47p4747", 46, 33, 24, 46, 19, "1111p11", 38 ],
        [ 31, 29, 27, 37, 36, 46, 33, 60, 60, 13, 12 ],
        [ 22, 52, 35, 38, 43, 23, 11, 32, 16, 15, 44, 25, 13 ],
        [ 29, 27, 38, 26, 19, 41, 46, 51, 26, 16, 17 ],
        [ 44, 32, 43, 29, 60, 60, 60, 35, 11, 13, 15 ],
        [ 26, 16, 41, 36, 21, 47, 17, 42, 35, 21, 19, 19, 53 ],
        [ 28, 37, 14, 27, 46, 34, 44, 33, 42 ],
        [ 41, 42, 47, 14, 60, 21, 21, "r44", 60 ],
        [ "和了", [ 0, 0, -3900, 4900 ],
        [ 3, 2, 3, "30符3飜3900点", "Pinfu(1飜)", "Riichi(1飜)", "Red Five(1飜)", "Ura Dora(0飜)" ] ] ],
        [ [ 5, 0, 0 ],
        [ 30500, 16000, 28700, 24800 ],
        [ 19 ],
        [],
        [ 37, 25, 27, 44, 28, 45, 29, 32, 34, 44, 52, 21, 28 ],
        [ 14, 38, 15, 12, "4444p44", 27, 33, 46, 38, 11, 17, 42, 13 ],
        [ 37, 60, 28, 21, 12, 45, 27, 60, 60, 60, 60, 60 ],
        [ 35, 24, 27, 31, 23, 18, 11, 26, 25, 34, 18, 41, 31 ],
        [ 34, 51, 26, 28, 44, 17, 39, 15, 24, 47, 11, 22 ],
        [ 41, 34, 31, 31, 60, 26, 60, 18, 24, 60, 18, 60 ],
        [ 14, 33, 22, 45, 41, 41, 28, 16, 26, 23, 37, 45, 16 ],
        [ 42, 17, 18, 19, "4141p41", 14, "45p4545", 22, 43, 32, 27, 33, 33 ],
        [ 60, 37, 33, 28, 26, 23, 22, 60, 60, 60, 60, 60, 60 ],
        [ 46, 38, 16, 12, 44, 37, 39, 47, 35, 21, 18, 29, 24 ],
        [ 26, 41, 29, 13, 29, 23, 43, 16, 46, 47, 42, 36, 43 ],
        [ 29, 21, 60, 41, 60, 46, 26, 47, 18, 60, 60, 43, 60 ],
        [ "和了", [ 2000, -1000, -500, -500 ],
        [ 0, 0, 0, "30符2飜500-1000点", "Seat Wind(1飜)", "Red Five(1飜)" ] ] ],
        [ [ 6, 0, 0 ],
        [ 32500, 15000, 28200, 24300 ],
        [ 18 ],
        [],
        [ 28, 44, 14, 23, 19, 14, 14, 38, 25, 38, 36, 45, 32 ],
        [ 39, 11, 22, 39, 26, 45, 34, 45, 42, 46, 33 ],
        [ 44, 60, 28, 25, 60, 32, 19, 38, 38, 60, 22 ],
        [ 27, 14, 42, 16, 22, 27, 37, 31, 17, 27, 21, 42, 41 ],
        [ 21, 39, 32, 42, 26, 26, 44, 18, 33, 25, 41 ],
        [ 31, 14, 60, 39, 37, 41, 16, 44, 21, 21, 22 ],
        [ 19, 22, 24, 32, 41, 13, 32, 21, 35, 47, 34, 22, 24 ],
        [ 31, 33, 11, 29, 47, 45, 25, 23, 34, 31, "2424p24", 52, 12 ],
        [ 47, 41, 19, 60, 60, 60, 32, 22, 25, 60, 35, 60 ],
        [ 18, 15, 36, 35, 16, 46, 12, 13, 34, 26, 21, 43, 36 ],
        [ 13, 27, 37, 28, 19, 16, 38, 38, 29, 24, 47, 47 ],
        [ 46, 21, 18, 12, 60, 60, "r43", 60, 60, 60, 60, 60 ],
        [ "和了", [ -500, -500, 2500, -500 ],
        [ 2, 2, 2, "30符1飜500点∀", "Mixed Triple Sequence(1飜)" ] ] ],
        [ [ 6, 1, 0 ],
        [ 32000, 14500, 30700, 22800 ],
        [ 39 ],
        [ 24 ],
        [ 22, 45, 45, 46, 19, 33, 45, 41, 23, 34, 11, 51, 24 ],
        [ 22, 21, 16, 36, "c222324", 34, 24, 15, 11, 34, 31, 25, 12, 15, 17, 15, 45, 31 ],
        [ 41, 46, 11, 60, 19, 60, 60, 60, 60, 60, 21, 60, 31, 22, 22, 45, 60, 45 ],
        [ 33, 35, 23, 38, 41, 17, 33, 37, 42, 47, 18, 46, 27 ],
        [ 17, 28, 26, 37, 43, 38, 27, 47, 36, 11, 47, 26, 25, 42, 32, 37, 13, 33 ],
        [ 41, 46, 42, 47, 60, 23, 28, 60, 18, 60, 36, "r47", 60, 60, 60, 60, 60, 60 ],
        [ 29, 13, 44, 28, 22, 52, 17, 14, 46, 12, 14, 11, 36 ],
        [ 39, 47, 13, 16, 19, 29, 12, 44, 28, "c181719", "c111213", 43, 19, 42, 42, 23, 18, 34, 39 ],
        [ 60, 44, 46, 22, 36, 52, 28, 60, 60, 47, 29, 29, 43, 60, 60, 60, 19, 60, 60 ],
        [ 22, 38, 21, 46, 18, 37, 43, 12, 14, 25, 35, 32, 44 ],
        [ 13, 16, 32, 27, 27, 43, 41, 32, 26, 44, 21, 31, 24, 21, 41, 31, 29, 19 ],
        [ 43, 46, 35, 21, 22, 44, 60, 43, 18, 16, 60, 44, "r31", 60, 60, 60, 60, 60 ],
        [ "和了", [ 0, 0, -5500, 7500 ],
        [ 3, 2, 3, "40符3飜5200点", "Riichi(1飜)", "Under the River(1飜)", "Ura Dora(1飜)" ] ] ],
        [ [ 7, 0, 0 ],
        [ 32000, 13500, 25200, 29300 ],
        [ 45, 38 ],
        [],
        [ 33, 22, 19, 15, 41, 11, 31, 32, 14, 19, 37, 25, 45 ],
        [ 43, 42, 39, 42, 23, 24, 39, 24, 36, 37, 44, 25, "c161415", 32, 47, 24, 12, 26 ],
        [ 41, 43, 11, 22, 39, 37, 60, 24, 60, 60, 60, 25, 45, 32, 60, 24, 60, 60 ],
        [ 17, 46, 13, 12, 35, 28, 22, 22, 41, 29, 17, 12, 46 ],
        [ 22, 51, 11, "m22222222", 33, 41, 32, 44, 17, 14, 36, 14, 26, 27, 39, 19, 47, "p121212", 18 ],
        [ 28, 41, 29, 0, 11, 60, 35, 60, 32, 33, 60, 13, 60, 51, 27, 39, 19, 47, 60 ],
        [ 17, 11, 31, 44, 34, 19, 29, 23, 45, 36, 27, 34, 47 ],
        [ 21, 21, 39, 52, 37, 44, 33, 15, 28, 13, 12, 46, 46, 18, 25, 35, 14, 29 ],
        [ 31, 47, 45, 11, 39, 29, 21, 21, 33, 36, 13, 52, 15, 44, 44, 19, 12, 18 ],
        [ 18, 24, 38, 31, 15, 34, 31, 42, 36, 53, 38, 35, 42 ],
        [ 37, 23, 23, 38, 21, 26, 43, 43, 47, "c333453", 28, 43, 16, 34, 16, 28, 13, 21, 16 ],
        [ 18, 15, 60, 31, 60, 60, 60, 60, 60, 24, 60, 23, 60, 43, 60, 60, 60, 60, 31 ],
        [ "Ryuukyoku", [ 1500, 1500, -1500, -1500 ] ] ] ],
`
Each line follows the instructions in this file: https://gist.githubusercontent.com/Equim-chan/875a232a2c1d31181df8b3a8704c3112/raw/a0533ae7a0ab0158ca9ad9771663e94b82b61572/downloadlogs.js

In short, the [0, 0, 0] represents the index of 局 (4*n+i), 本场 立直棒.
Next line represents 四家点数, followed by 宝牌、里宝牌.
Next three lines represents each players' 配牌(haipai) 进张(zimo hai) 舍张(kiri hai)
The last line figures out [和牌者，放铳人/自摸方，和牌者/包牌方]

Here is the issue: **To facilitate the use of GET, one might prefer to get the list of four players score condition**
